pkg/prometheus/operator.go

```diff
@@ -2157,3 +2163,11 @@ func validateRelabelConfig(rc monitoringv1.RelabelConfig) error {
 	}
 	return nil
 }
+
+func validateProberURL(url string) bool {
```

Comment:
IMHO it would be better if `validateProberURL` returns an error. This way the returned message would be clearer (e.g. "invalid host" vs. "invalid port") and it would be consistent with the other validate*() functions.

Reply:
I actually wrote with error as return but found it a bit tricky when checked with different types of inputs. Like the case url passed is `http://blackbox-exporter.example.com:8080` in this case when we split the string on `:` it will take `/blackbox-exporter.example.com` as hostPort[1] and if we validate and return error as invalid port it could be more confusing